cli/tools/upgrade.rs

```diff
@@ -41,6 +41,16 @@ pub async fn upgrade_command(
 
   let install_version = match version {
     Some(passed_version) => {
+      if canary
+        && !regex::Regex::new("^[0-9a-f]{40}$")?.is_match(&passed_version)
```

Comment:
maybe create the regexp at top level
```rust
lazy_static! {
  static ref VERSION_STRING_RE: Regex = regex::Regex::new("^[0-9a-f]{40}$").unwrap();
}

Reply:
for what reason? this is used only there

Reply:
because it is [recommended](https://github.com/rust-lang/regex/blob/master/PERFORMANCE.md) by regexp-rs and is consistent with rest of the codebase.

- https://github.com/denoland/deno/blob/f46e39c5c5435696daab9df075482e6e5c83533b/runtime/colors.rs
- https://github.com/denoland/deno/blob/8708d3c0451129792d58b7fa856101ceaa7bf487/test_util/src/lsp.rs

Reply:
It says in the scenario of a loop/it being called multiple times. in that case it makes sense, but in the case here, this regex is executed only once ever. changing it to lazy_static would have no performance benefit

Reply:
Yeah, inlining it is fine as it will only be executed once.

Reply:
sounds good 👍